### PR TITLE
Refactoring and more unit tests

### PR DESCRIPTION
**Pull request by bricoletc**
TLDR: shorter and more readable code, and unit tests on the individual functions

My initial main goal was to enforce the following: **the first sequence given in the input MSA should be the first path through the output PRG**

This is currently a strong assumption `gramtools` makes, for genotyping.
I don't know if this codebase was initially designed this way. I did find it was not enforced on master branch. I fixed it and added quite a few tests that give me more confidence it is now enforced. It might be more tests are needed to check it holds for nested prgs later.

In the process, I refactored the code. I made an explicit effort to i) Reduce if/else induced code nesting and ii) Make the code shorter, by for eg removing assertions (with tests) and terser code constructs. I find the code easier to delve into now.

I also added more general unit tests which complement the existing integration tests by testing the individual functions used by `make_prg_from_msa` module.

Also fixes bug #12

### CODE REVIEW

**Comment by rmcolq** [comment]
Sounds excellent. Originally the code was written to preserve first sequence as reference. This has been lost during refactoring because it was not necessary for `pandora`.

**Comment by bricoletc** [comment]
I had a feeling it was, could see several features which looked exactly like that!
